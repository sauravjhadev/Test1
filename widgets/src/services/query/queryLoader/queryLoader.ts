import * as fs from 'fs';

import WidgetFileType, { FileLoaderResponse } from '../../../types';
import { messages } from '../../../messages';

export default function queryLoader(widgetDir: string): Promise<FileLoaderResponse> {
    return new Promise((resolve, reject) => {
        fs.readFile(
            `${widgetDir}/${WidgetFileType.QUERY}`,
            'utf8',
            (error, data: string) => {
                if (error) {
                    if (error.code === 'ENOENT') {
                        resolve({
                            type: WidgetFileType.QUERY,
                            data: '',
                        });
                    } else {
                        reject(messages.invalidQuery());
                    }
                }

                resolve({
                    type: WidgetFileType.QUERY,
                    data,
                });
            },
        );
    });
}
