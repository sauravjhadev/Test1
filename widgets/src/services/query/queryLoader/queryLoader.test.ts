import queryLoader from './queryLoader';

describe('queryLoader', () => {
    it('correctly loads query', (done) => {
        queryLoader('src/services/__fixtures__').then((data) => {
            expect(data).toMatchSnapshot();
            done();
        });
    });
});
