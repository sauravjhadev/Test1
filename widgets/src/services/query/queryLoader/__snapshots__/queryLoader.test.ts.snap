// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['queryLoader correctly loads query 1'] = `
Object {
  "data": "query ProductList($productIds: [Int!]) {
    site {
        settings {
            storeName
        }
        products(entityIds: $productIds) {
            edges {
                node {
                    name
                    entityId
                    addToCartUrl
                    addToWishlistUrl
                    path
                    sku
                    prices {
                        price {
                            currencyCode
                            value
                        }
                    }
                    reviewSummary {
                        summationOfRatings
                        numberOfReviews
                    }
                    defaultImage {
                        url(width: 1500, height: 1500)
                    }
                }
            }
        }
    }
}
",
  "type": "query.graphql",
}
`;
